From: Jan de Mooij <jdemooij@mozilla.com>
Date: Wed, 23 Sep 2020 08:22:28 +0100
Subject: Bug 1666646 - Set CodeAlignment to 16 for the 'none' backend.
 r=jonco

This fixes a static_assert on 32-bit platforms because 4 is too small for the JitCode
cell header alignment.

The CodeAlignment value does't really matter as the none backend does not allocate
any JIT code or JitCode instances, but 16 matches x86 and x64.

Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1666646
Forwarded: https://phabricator.services.mozilla.com/D91093
Applied-upstream: 83, commit:https://hg.mozilla.org/mozilla-central/rev/f7e1d7c2fab7
---
 js/src/jit/none/MacroAssembler-none.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/src/jit/none/MacroAssembler-none.h b/js/src/jit/none/MacroAssembler-none.h
index c108e7b..4798811 100644
--- a/js/src/jit/none/MacroAssembler-none.h
+++ b/js/src/jit/none/MacroAssembler-none.h
@@ -100,7 +100,7 @@ static constexpr Register WasmTlsReg{Registers::invalid_reg};
 static constexpr Register WasmJitEntryReturnScratch{Registers::invalid_reg};
 
 static constexpr uint32_t ABIStackAlignment = 4;
-static constexpr uint32_t CodeAlignment = sizeof(void*);
+static constexpr uint32_t CodeAlignment = 16;
 static constexpr uint32_t JitStackAlignment = 8;
 static constexpr uint32_t JitStackValueAlignment =
     JitStackAlignment / sizeof(Value);
